src/utils/headerInterpreter.ts

```diff
@@ -0,0 +1,47 @@
+import { Header, HeadersInterpreter } from "axios-cache-interceptor"
+import { parse } from "cache-parser"
+
+// NOTE: Taken with reference to https://github.com/arthurfiorette/axios-cache-interceptor/blob/v0.9.2/src/header/interpreter.ts
+// The change made is to remove the `maxAge` condition, which is not suitable because
+// we want to revalidate on update.
+// An alternative is to invalidate the cache on `update` but
+// this requires an exhaustive search through our codebase.
+// This incurs an extra network call on every update, which is not ideal but in 304
+// this is an empty request.
+// eslint-disable-next-line import/prefer-default-export
+export const headerInterpreter: HeadersInterpreter = (headers) => {
+  if (!headers) return "not enough headers"
+
+  const cacheControl = headers[Header.CacheControl]
+
+  if (cacheControl) {
+    const { noCache, noStore, mustRevalidate, maxAge, immutable } = parse(
+      String(cacheControl)
+    )
+
+    // Header told that this response should not be cached.
+    if (noCache || noStore) {
+      return "dont cache"
+    }
+
+    if (immutable) {
+      // 1 year is sufficient, as Infinity may cause more problems.
+      // It might not be the best way, but a year is better than none.
+      return 1000 * 60 * 60 * 24 * 365
+    }
+
+    // Already out of date, for cache can be saved, but must be requested again
+    if (mustRevalidate || maxAge) {
```

Comment:
Hmm ok the [docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) say that the must-revalidate response directive indicates that the response can be stored in caches and can be reused while [fresh](https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching#fresh_and_stale_based_on_age). If the response becomes [stale](https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching#fresh_and_stale_based_on_age), it must be validated with the origin server before reuse. 

Does this not mean that for all practical purposes, we need to treat age>max_age as `mustRevalidate`?

> in our case, only the maxAge header is present, and the mustRevalidate header is missing

just to clarity, are you saying that github's responses never has the mustRevalidate header? if that is the case, why are we checking for the existence of `mustRevalidate` ah? Additionally, does Github's response always have a max header of 60s?

Reply:
for clarity, the headers are returned by github to us, and they serve as a guideline for how we can cache the requests. 

> Does this not mean that for all practical purposes, we need to treat age>max_age as `mustRevalidate`?

yeap, precisely why we return `0` when the `maxAge` header is present! we want to check with github whether the data we have cached is still fresh.

> just to clarity, are you saying that github's responses never has the mustRevalidate header? if that is the case, why are we checking for the existence of mustRevalidate ah? Additionally, does Github's response always have a max header of 60s?

github's example [response](https://docs.github.com/en/rest/guides/getting-started-with-the-rest-api?apiVersion=2022-11-28#about-the-response-code-and-headers) and the few i checked do not have `mustRevalidate` - but this does **not** imply absence of `mustRevalidate`. because we can only say with confidence for a few subsets of github's API, we should still interpret the headers in decreasing order of strictness.